styles/main.css

```diff
@@ -158,11 +158,11 @@ a[href="#top"] {
     margin-left: -25px;
     bottom: 10px;
     display: inline-block;
+    transition: 0.5s;
 }
 a[href="#top"]:hover {
     text-decoration: none;
     transform: rotate(45deg);
-    transition: 0.5s;
```

Comment:
Put transitions in the 'normal state' rule so that the animation happens in both directions